shared-unpackaged/is-obscured.js

```diff
@@ -26,55 +34,70 @@ export default function isObscured(element: Element): boolean {
+    // viewport is zoomed such that both corners are off screen but the rest
+    // isn't. In this case some browsers don't return the element from
+    // `elementFromPoint` then doesn't return the element.
+    // Also, consider how we might mitigate the pointer-events issue and make
+    // this call more robust.
+    const bounds = getBounds(anchorElement);
+
+    // This method does the main work, taking some coordinates and determining
+    // if our element is visible at that point or not.
+    const isVisible = (x: number, y: number): boolean => {
+        const elAtPoint = document.elementFromPoint(x, y);
+        if (
+            elAtPoint != null &&
+            (elAtPoint === popperElement || popperElement.contains(elAtPoint))
+        ) {
+            // Oh no, we're being obscured by our own popper.
+            // We need to look behind it. Shenanigans time.
+            const pointerEventsStyle = elAtPoint.style.pointerEvents;
+            // Remove pointer events so that we can look through it.
+            elAtPoint.style.pointerEvents = "none";
+            try {
+                const visible = isVisible(x, y);
+                return visible;
+            } finally {
+                // Make sure we put things back the way we found them. :)
+                elAtPoint.style.pointerEvents = pointerEventsStyle;
+            }
+        }
+        if (anchorElement instanceof Element) {
+            // If we are working with an element, then we can do some decendency checks
+            // to ensure we're not just hitting a child. We're ok with saying that
+            // we're visible if we hit a parent because we check them for visibility
+            // elsewhere.
+            return (
+                elAtPoint != null &&
+                (anchorElement.contains(elAtPoint) ||
+                    elAtPoint.contains(anchorElement))
+            );
+        }
+
+        // If element is a reference object, all we have to work with is
+        // intersection for checking obscurity. Since this doesn't cover
+        // parent/child relationships in the DOM, it's not really effective
+        // on its own and is possibly about as good as just returning `true`.
+        const intersection =
+            elAtPoint && getIntersection(bounds, getBounds(elAtPoint, true));
+        return (
+            intersection?.horizontal !== "within" ||
+            intersection?.vertical !== "within"
+        );
+    };
+
+    // NOTE: We are using functions here so that we only do as much work
+    // as we need to, short-circuiting as soon as we have a definitive
+    // answer.
+    const isTopLeftVisible = () => isVisible(bounds.left, bounds.top);
+    const isBottomRightVisible = () => isVisible(bounds.right, bounds.bottom);
+    const isCenterVisible = () =>
+        isVisible(
+            bounds.left + (bounds.right - bounds.left) / 2,
+            bounds.top + (bounds.bottom - bounds.top) / 2,
+        );
```

Comment:
Added a check for the center of the anchor too, to improve this check a bit. Also, made these function calls so that we only execute as much code as we need to, which should speed this check up in many cases.